### ISSUE TITLE
Connection gets closed because the "variant" data type is not supported

### ISSUE THREAD

**Issue by sualeh** [Microsoft/mssql-jdbc#91]
When I execute `SELECT * FROM INFORMATION_SCHEMA.SEQUENCES`, the connection gets closed, with an exception indicating that the "variant" data type is not supported. Consider throwing `SQLServerException` with a state other than `com.microsoft.sqlserver.jdbc.SQLServerException.EXCEPTION_XOPEN_CONNECTION_FAILURE`, and not closing the database connection.

Please let me know if you need more information.

**Comment by sualeh** [comment]
Partial stack trace:

```
com.microsoft.sqlserver.jdbc.SQLServerException: The "variant" data type is not supported.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:206)
	at com.microsoft.sqlserver.jdbc.TypeInfo$Builder$16.apply(dtv.java:3198)
	at com.microsoft.sqlserver.jdbc.TypeInfo$Builder.build(dtv.java:3360)
	at com.microsoft.sqlserver.jdbc.TypeInfo.getInstance(dtv.java:3426)
	at com.microsoft.sqlserver.jdbc.StreamColumns.setFromTDS(StreamColumns.java:217)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet$1CursorInitializer.onColMetaData(SQLServerResultSet.java:240)
	at com.microsoft.sqlserver.jdbc.TDSParser.parse(tdsparser.java:99)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.<init>(SQLServerResultSet.java:335)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet42.<init>(SQLServerResultSet42.java:35)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1684)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.doExecuteStatement(SQLServerStatement.java:903)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement$StmtExecCmd.doExecute(SQLServerStatement.java:796)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7535)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:2438)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:208)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:183)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.execute(SQLServerStatement.java:769)
	at ...
```

**Comment by v-suhame** [comment]
Thanks @sualeh! We are looking into this issue. If the driver is to leave the connection open after throwing appropriate exception, it should discard the TDS packet/packets from the response, we are looking for an effective way to do that. Do let us know if you have any suggestions for that.

**Comment by v-suhame** [comment]
Here is the [PR](https://github.com/Microsoft/mssql-jdbc/pull/99) with fix for connection close issue. Please use it for testing and update us. Thanks!

**Comment by v-nisidh** [comment]
Fixed in [6.1.2 ](https://github.com/Microsoft/mssql-jdbc/tree/v6.1.2)
